### PR TITLE
Add S3 Scheme as valid URL Scheme

### PR DESCRIPTION
**Pull request by stefanbenten**
The AWS CLI forces to use the s3:// scheme which did not work with the latest uplink updates.
This PR should add it back to the list of allowed schemes.

### CODE REVIEW

**Comment by stefanbenten** [comment]
Seems good besides the kademlia tests that are failing

File: internal/fpath/path.go [hunk 1]
```diff
@@ -12,7 +12,10 @@ import (
 	"strings"
 )
 
-const storjScheme = "sj"
+var storjScheme = map[string]bool{
```
Comment (kaloyan-raev) [inline internal/fpath/path.go]: This works, but perhaps you want to make it even better by declaring it as `map[string]struct{}`, as described in this article: https://emersion.fr/blog/2017/sets-in-go/

Comment (stefanbenten) [reply internal/fpath/path.go]: Will look into that!

File: internal/fpath/path.go [hunk 2]
```diff
@@ -12,7 +12,11 @@ import (
 	"strings"
 )
 
-const storjScheme = "sj"
+// Create a set
+var storjScheme = map[string]struct{}{
```
Comment (coyle) [inline internal/fpath/path.go]: just curious why `map[string]struct` ?

Comment (coyle) [reply internal/fpath/path.go]: not blocking btw

Comment (stefanbenten) [reply internal/fpath/path.go]: It is saving resources and improves performance. Kaloyan left a good post: https://emersion.fr/blog/2017/sets-in-go/

Comment (stefanbenten) [reply internal/fpath/path.go]: Instead of having a byte for an bool

Comment (coyle) [reply internal/fpath/path.go]: nice! I've always just used `map[string]bool` but will now be switching to `map[string]struct` great write up @kaloyan-raev

**Pull request merged**